appeals/web/src/server/lib/mappers/appeal/appeal.mapper.js

```diff
@@ -0,0 +1,139 @@
+import { generateIssueDecisionUrl } from '#appeals/appeal-details/issue-decision/issue-decision.mapper.js';
+import {
+	mapDocumentDownloadUrl,
+	mapVirusCheckStatus
+} from '#appeals/appeal-documents/appeal-documents.mapper.js';
+import { APPEAL_CASE_STATUS, APPEAL_VIRUS_CHECK_STATUS } from 'pins-data-model';
+import { conjoinedPromises } from '#common/conjoined-promises.js';
+import { submaps as hasSubmaps } from './has.js';
+import { submaps as s78Submaps } from './fpa.js';
+
+/**
+ * @typedef {import('#appeals/appeal-details/appeal-details.types.js').WebAppeal} WebAppeal
+ * @typedef {import('../../../app/auth/auth-session.service.js').SessionWithAuth} SessionWithAuth
+ */
+
+/**
+ * @typedef {(
+ *   appealDetails: WebAppeal,
+ *   currentRoute: string,
+ *   session: SessionWithAuth,
+ *   skipAssignedUsersData?: boolean
+ * ) => Promise<{appeal: MappedInstructions}>} Mapper
+ */
+
+/**
+ * @typedef {(
+ *   appealDetails: WebAppeal,
+ *   currentRoute: string,
+ *   session: SessionWithAuth,
+ *   skipAssignedUsersData?: boolean
+ * ) => Promise<Instructions>} SubMapper
+ */
+
+// Not super buzzed about these keys, is there something nicer than appealType we can use here?
+/** @type {Record<string, [string, SubMapper][]>} */
+const submaps = {
+	Householder: hasSubmaps,
+	'Planning appeal': s78Submaps
+};
+
+/** @type {Mapper} */
+export async function initialiseAndMapAppealData(...args) {
+	const [appealDetails] = args;
+
+	if (appealDetails === undefined) {
+		throw new Error('appealDetails is undefined');
+	}
+
+	if (!appealDetails.appealType) {
+		throw new Error('No appealType on appealDetails');
+	}
+
+	const submap = submaps[appealDetails.appealType];
+
+	const submappedData = await conjoinedPromises(submap, ([, submapper]) => submapper(...args));
+
+	return Array.from(submappedData).reduce(
```

Comment:
Again it's all subjective, but for me personally, this code is quite difficult to read and get an understanding of the intent. Is there any way this could be refactored to improve clarity?